src/docsfy/postprocess.py

```
@@ -15,12 +16,168 @@
+                # Rewrite to canonical slug casing for case-sensitive hosts
+                canonical = slug_to_canonical.get(target_lower, target_slug)
+                if canonical != target_slug:
+                    # Preserve any anchor/title suffix from the original match
+                    full_match = match.group(0)
+                    return full_match.replace(
+                        f"({target_slug}.html", f"({canonical}.html"
+                    )
+                return match.group(0)  # Already correct casing
+            logger.info(
+                f"[{_label}] Removing broken link to '{target_slug}.html' "
+                f"in page '{_slug}'"
```

Ruff (0.15.10)

[warning] Logging statement uses f-string

(G004)